editor/components/block-list/block.js

```diff
@@ -439,6 +440,7 @@ export class BlockListBlock extends Component {
 					{ showUI && <BlockMobileToolbar uid={ block.uid } /> }
 				</div>
 				{ !! error && <BlockCrashWarning /> }
+				<BlockListSiblingInserter uid={ block.uid } />
```

Comment:
I am not certain moving sibling inserter inside block-list will not have the undesirable side effect of making sibling inserter unusable for locked blocks cc: @gziolo.

Reply:
What do you mean by "locked blocks"?

Reply:
I guess you mean "template locking". In which case, this change doesn't change the current behavior which is: The `Inserter` component does not show anything if the template is locked. This component is included in the `BlockListSiblingInserter` component.

Reply:
In PR https://github.com/WordPress/gutenberg/pull/3493 BlockListBlock was made extensible to allow the creation of Frozen/locked blocks. Blocks that are temporarily not editable because of co-editing for example. It looks like the hook allows us to "change" BlockListBlock, so in places that use this hook, we may be disabling sibling inserter.

Reply:
Good point! If that means the hook should be "smarter" or the hook should target `BlockEdit` instead?

Reply:
In accordance with the conversation on PR https://github.com/WordPress/gutenberg/pull/3493 adding the hook to BlockEdit does not fit the needs, because it would not be possible to lock the toolbar or the block transform mechanism. So we can make the hook smarter or we can try to add a component that wraps BlockListBlock and the sibling inserter.

Reply:
@jorgefilipecosta thanks for raising this issue. Yes, it might be a bit inconvenient, but I'm sure we can find workarounds :) Don't treat it as a blocker as we can figure out solutions later.

> So we can make the hook smarter or we can try to add a component that wraps BlockListBlock and the sibling inserter.

It seems like a wrapping div would work perfectly fine. I like the idea. I didn't think about it too much though :)